src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -720,6 +728,46 @@ func (r *ClusterReconciler) reconcileClusterForPods(pod client.Object) []reconci
 	}
 }
 
+func (r *ClusterReconciler) reconcileClusterForExternalCASecret(s client.Object) []reconcile.Request {
+	hasExternalCA, found := s.GetAnnotations()[SecretAnnotationExternalCAKey]
+	if !found || hasExternalCA != "true" {
+		return nil
+	}
+
+	clusterName, found := s.GetLabels()[labels.InstanceKey]
+	if !found {
+		return nil
+	}
+
+	clusterList := &redpandav1alpha1.ClusterList{}
+	err := retry.OnError(retry.DefaultRetry,
+		func(err error) bool {
+			return err != nil
+		},
+		func() error {
+			return r.List(context.TODO(), clusterList)
+		})
```

Comment:
Why mapping function is doing any network call e.g. listing cluster CR?

REF
https://github.com/kubernetes-sigs/controller-runtime/blob/d989e66740a55ab67822a6abd8fa595455d70315/pkg/handler/enqueue_mapped.go#L28-L46

Reply:
The purpose of the listing is to find the namespace of the cluster, and check that the cluster exists so that the controller does not receive request for a cluster that does not exist. We could add an annotation that specifies the namespace, then we would call Get() instead of List().

Reply:
This is anti-pattern. 

There is similar functionality within Reconcile function https://github.com/redpanda-data/redpanda/blob/a628e95cf23f1d60a29b2d7313f688ca975cac74/src/go/k8s/controllers/redpanda/cluster_controller.go#L113-L123. To not duplicate functionality I would add cluster resource to `reconcile.Request{}`

Another question is how the label will be removed if it is available in the secret, but cluster is deleted?

Reply:
Agree. See https://github.com/redpanda-data/redpanda/pull/9863#discussion_r1168709500